.github/workflows/deployment.yml

```diff
@@ -0,0 +1,43 @@
+name: Initiate Dev or Prod Environmnet Deployment
+
+on:
+    push:
+       branches:
+          - master
+          - production
+env:
+    DEV_VM_IP: '${{secrets.DEV_VM_IP}}'
+    DEV_VM_PSWD: '${{secrets.DEV_VM_PSWD}}'
+    PROD_VM_IP: '${{secrets.PROD_VM_IP}}'
+    PROD_VM_PSWD: '${{secrets.PROD_VM_PSWD}}'
+    DISCORD_WH: '${{secrets.DEPLOYMENT_CHANNEL_WEBHOOK}}'
+jobs:
+    deploy:
+        runs-on: ubuntu-latest
+        strategy:
+            matrix:
+                python-version: ["3.10"]
+
+        steps:
+            - uses: actions/checkout@v3
+
+            - name: Set up Python ${{ matrix.python-version }}
+              uses: actions/setup-python@v4
+              with:
+                python-version: ${{ matrix.python-version }}
+
+            - name: Install dependencies
+              run: |
+                python3 -m pip install --upgrade pip
+                pip3 install paramiko
+
+            - name: Extract branch name
+              id: extract_branch_name
+              run: echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> $GITHUB_OUTPUT
+
+            - name: Start deployment
+              run: |
+                if [ "${{ steps.extract_branch_name.outputs.branch }}" == "master" ]; then
```

Comment:
As of now, this will only run on production and master pushes. I believe the first conditional check should be if production and instead of using `elif` use `else`.

Reply:
@asynchroza For the else I understand, but why the first should be production, as we will first push to master?

Reply:
This leaves room for us to further extend the logic for dev deploying from different branches. Production is fixed and it's not expected to be changed.
We don't want to have the production check in an else block.